libr/debug/p/native/linux/linux_debug.c

```diff
@@ -517,3 +521,1439 @@ RList *linux_desc_list (int pid) {
 	closedir (dd);
 	return ret;
 }
+
+
+/* Coredump  functions */
+
+static map_file_t mapping_file = {0,0};
+
+static char *prpsinfo_get_fname(char *buffer) {
+	char *p;
+	
+	/* buffer contains: str\0str1\0str2\0... Here we're only interested in the first part of string, and strdup copies till it reaches a \0 */
+	p = strdup (buffer);
+	return p;
+}
+
+static char *get_basename(char *pfname, int len) {
+	char *p;
+
+	for (p = pfname + len; p != pfname ; p--) {
+		if (*p == '/') return (p + 1);
+	}
+
+	return p;
+}
+
+static char *prpsinfo_get_psargs(char *buffer, char *pfname, int size_psargs, int len) {
+	char paux[ELF_PRARGSZ];
```

Comment:
Lgtm

> On 20 May 2016, at 11:30, leberus <notifications@github.com> wrote:
> 
> In libr/debug/p/native/linux/linux_debug.c:
> 
> > +	p = strdup (buffer);
> > +	return p;
> > +}
> > +
> > +static char *get_basename(char *pfname, int len) {
> > +	char *p;
> > +
> > +	for (p = pfname + len; p != pfname ; p--) {
> > +		if (*p == '/') return (p + 1);
> > +	}
> > +
> > +	return p;
> > +}
> > +
> > +static char *prpsinfo_get_psargs(char *buffer, char *pfname, int size_psargs, int len) {
> > +	char paux[ELF_PRARGSZ];
> This is just 80bytes
> 
> x86_64-linux-gnu/sys/procfs.h:#define ELF_PRARGSZ (80)
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
>